STYLEGUIDE.md

```diff
@@ -0,0 +1,716 @@
+const foo = "foo";
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use single quotes */
+const foo = 'foo';
+
+/* Use double quotes if the quoted string contains single quotes. */
+const foo = "'foo'";
+```
+
+### String concatenation
+When possible use template literals, this makes the code more readable in most cases.
+Feel free to concatenate strings with `+` in case it makes to code better readable.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't concatenate strings with `+` */
+const foo = 'foo ' + bar + ' bar';
+
+/* Grey area: If template literals become too complex, don't use them. */
+const foo = `foo ${fn(`${bar}bar`)}`;
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Concatenate strings using template literals */
+const foo = `foo ${bar} bar`;
+
+/* Grey area: If template literals become too complex, feel free to concatenate with +. */
+const foo = `foo ${fn(bar +'bar')}`;
+```
+
+## Arrays
+
+### Use `.forEach` rather than `for`
+Use the functional language concepts of JavaScript in favor of the imperative ones, when possible.
+Use the correct function for your use case.
+In rare cases, when performance is of the essence, it makes sense to resort to the imperative `for` instead.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use for ... of if you can use .forEach */
+for (const item of array) {
+  ...
+}
+
+/* Don't use .forEach if you can use .map */
+const result = [];
+array.forEach(item => {
+  result.push(item.property);
+});
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use .forEach if possible */
+array.forEach(item => {
+  ...
+});
+
+/* Use .map, if appropriate */
+const result = array.map(item => item.property);
+```
+
+
+## `null` and `undefined`
+
+### Use optional parameters for optional values
+Use optional parameters, if there is a default value or if it semantically makes sense to omit the parameters when consuming the API.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use explicit undefined if the parameter is semantically optional */
+function foo(options: Options | undefined) {
+  ...
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use optional parameter if the parameter is semantically optional */
+function foo(options?: Options) {
+  ...
+}
+
+/* Use default parameter if there is a default (making the parameter inherently optional for the consumer) */
+function foo(options: Options = {}) {
+  ...
+}
+```
+
+### Use `undefined` for explicit absence of values
+If a value semantically is not optional prefer using `undefined` for values that do not exist or are not defined.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use null for non-existent values */
+function foo(obj: SomeType | null) {
+  ...
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use undefined for non-existent values */
+function foo(obj: SomeType | undefined) {
+  ...
+}
+```
+
+### Use `null` for intentional explicit absence of values or if it is part of an external API
```

Comment:
```suggestion
### Use `null` for the intentional explicit absence of values or if it is part of an external API
```

Reply:
I think the correct form is without the because this is a non-count word. I googled that and found english teacher melanie to the rescue: http://www.englishteachermelanie.com/grammar-when-not-to-use-the-definite-article/